test/dynamo/test_subclasses.py

```diff
@@ -522,6 +522,22 @@ def fn(x):
         res, _ = fn(input)
         self.assertFalse(res)
 
+    def test_disable_all_torch_function(self):
+        @torch.compile(backend="eager")
+        def fn(x):
+            with torch._C.DisableTorchFunction():
+                torch._dynamo.graph_break()
+                return (
+                    torch._C._is_torch_function_enabled(),
+                    torch._C._is_torch_function_all_disabled(),
+                    torch.add(x, 1.0),
+                )
```

Comment:
Per https://github.com/pytorch/pytorch/pull/149489/files#r2003640222, what if we have a test that
1. removes the graph break
2. puts the return outside after the with block.

Reply:
Added, although I left the graph break because it makes sure state still properly gets restored.

Reply:
Added both actually